### ISSUE TITLE
Terminal local `tsh login`

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#5288]
###

When using multiple terminal tabs, users do this

```
tsh login --proxy=x --cluster=a
```

Then in another window, they login into another account:

```
tsh login --proxy=y --cluster=z
```

The second command will switch the first tab to point to the proxy y and cluster z.

## Proposal

Add `non-sticky` login flag:

```
tsh login -n --proxy=x --cluster=a
```

When executed in non-sticky mode, tsh will:

* Use the parent PID of the process that launched tsh (equivalent of bash `$$`) and create a profile:

```
~/.tsh/profile-<pid>/profile.yaml
```

When this profile exists, subsequent shell executions by the same shell will refer to the profile `~/.tsh/profile-<pid>/profile.yaml`
```

**Comment by russjones** [comment]
An alternative proposal is to add a `tsh login` sub-command that prints `TELEPORT_PROXY` and `TELEPORT_CLUSTER` to stdout. The caller can `eval` the output and then all future `tsh ssh` commands will connect to that specific cluster.

This is a fairly common approach for Unix programs For example, below is how you start a SSH agent.

```
$ eval `ssh-agent`
Agent pid 175774
```

The above can alternatively be written as `eval $(ssh-agent).`

To make a Teleport session sticky to the terminal, we could add a `tsh login --env` flag like below.
```
$ eval `tsh login leaf.example.com --env`
> Profile URL:        https://proxy.example.com:3080
  Logged in as:       rjones
  Cluster:            leaf.example.com
  Roles:              admin
  Logins:             root
  Kubernetes:         disabled
  Valid until:        2021-01-14 09:27:07 +0000 UTC [valid for 7h59m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty
```

Even the output the user sees after `tsh login` will not change, because `tsh login --env` will output the following:

```
TELEPORT_PROXY=proxy.example.com:3080;
TELEPORT_CLUSTER=leaf.example.com;
echo "> Profile URL:        https://proxy.example.com:3080"
echo "Logged in as:       rjones"
echo "Cluster:            leaf.example.com"
echo "Roles:              admin"
echo "Logins:             root"
echo "Kubernetes:         disabled"
echo "Valid until:        2021-01-14 09:27:07 +0000 UTC [valid for 7h59m0s]"
echo "Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty"
```

After the above is run all `tsh ssh` in that terminal window will connect to the leaf cluster instead of the root cluster.

**Comment by a-palchikov** [comment]
Agree with @russjones - a much simpler and non-intrusive approach!

**Comment by russjones** [comment]
@klizhentas Discussed this ticket with the customer and want to pin the cluster and proxy. So instead of adding a flag to `tsh login`, I think the UX can be improved if we add a `tsh env` subcommand that will print out the configuration from the current profile.

This also matches the UX of Database Access `tsh db env`.

Here are a few examples:

```
$ tsh --proxy=example.com login
[...]
$ tsh env
export TELEPORT_PROXY=example.com
export TELEPORT_CLUSTER=root
```
```
$ tsh login leaf
[...]
# tsh env
export TELEPORT_PROXY=example.com
export TELEPORT_CLUSTER=leaf
```
```
$ tsh --proxy=remote.example.com login
[...]
$ tsh env
export TELEPORT_PROXY=remote.example.com
export TELEPORT_CLUSTER=root
```

**Comment by webvictim** [comment]
@russjones Did this get implemented already in #5395?
